### PR TITLE
Add back types to preset-env

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
> NOTE: This PR isn't targeting master; it's targeting update-to-core-js-3

@existentialism If you want to help you can push here, just leave a comment. We'll merge this PR to the main PR (#7646) when we are done.

- [x] ./available-plugins.js
- [x] ./filter-items.js
- [x] ./index.js
- [x] ./get-option-specific-excludes.js
- [x] ./debug.js
- [x] ./types.js
- [x] ./utils.js
- [x] ./options.js
- [x] ./targets-parser.js
- [x] ./normalize-options.js
- [x] ./module-transformations.js

Maybe not needed:
- [x] ./polyfills/regenerator/entry-plugin.js
- [x] ./polyfills/regenerator/usage-plugin.js
- [x] ./polyfills/corejs2/entry-plugin.js
- [x] ./polyfills/corejs2/get-platform-specific-default.js
- [x] ./polyfills/corejs2/built-in-definitions.js
- [x] ./polyfills/corejs2/usage-plugin.js
- [x] ./polyfills/corejs3/entry-plugin.js
- [x] ./polyfills/corejs3/shipped-proposals.js
- [x] ./polyfills/corejs3/built-in-definitions.js
- [x] ./polyfills/corejs3/usage-plugin.js

### CODE REVIEW

File: packages/babel-preset-env/src/types.js [hunk 1]
```diff
@@ -13,9 +13,11 @@ export type Targets = {
 export type ModuleOption = $Values<typeof ModulesOption>;
 export type BuiltInsOption = $Values<typeof UseBuiltInsOption>;
 
+type CorejsVersion = 2 | 3;
```
Comment (zloirock) [inline packages/babel-preset-env/src/types.js]: If can be also `false`, string and allow minor versions like `corejs: '3.1'`.

Comment (zloirock) [reply packages/babel-preset-env/src/types.js]: Moreother, finally it's `semver` instance.

**Comment by existentialism** [comment]
@nicolo-ribaudo working on normalize-options

**Comment by nicolo-ribaudo** [comment]
Do you think that the types in the internal plugins are needed?

File: packages/babel-preset-env/src/polyfills/corejs3/entry-plugin.js [hunk 2]
```diff
@@ -23,21 +29,30 @@ const BABEL_POLYFILL_DEPRECATION = `
   \`@babel/polyfill\` is deprecated. Please, use required parts of \`core-js\`
   and \`regenerator-runtime/runtime\` separately`;
 
+type Options = {
+  corejs: NormalizedCorejsOption,
+  include: Set<string>,
+  exclude: Set<string>,
+  polyfillTargets: Targets,
+  debug: boolean,
+};
```
Comment (zloirock) [inline packages/babel-preset-env/src/polyfills/corejs3/entry-plugin.js]: Maybe it would be better to move it to `src/types.js`?

**Pull request merged**